src/ray/core_worker/transport/dependency_resolver.cc

```diff
@@ -59,11 +59,15 @@ void InlineDependencies(
 
 void LocalDependencyResolver::CancelDependencyResolution(const TaskID &task_id) {
   absl::MutexLock lock(&mu_);
```

Comment:
Why changes in this file?

Reply:
this is the change Mengjin mentioned. Problem is we used TaskID to track dependency resolution requests. But at least in cpp test, a same taskID may call `ResolveDependencies` multiple times, a new request overrides and deletes an old dependency resolution request. But the test still assumes the old request finishes with callback called.

This PR fixes it by tracing each request not only by TaskID but also by an auto increment id.

I am not sure if the current behavior (forgetting old reqs) is expected, so I want some input from you.